src/Microsoft.VisualStudio.SDK.Analyzers/VSSDK004ProvideAutoLoadAttributeAnalyzer.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+
+namespace Microsoft.VisualStudio.SDK.Analyzers
+{
+    using System;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.VisualStudio.Shell;
+
+    /// <summary>
+    /// Discovers <see cref="ProvideAutoLoadAttribute"/> usages without BackgroundLoad flag.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class VSSDK004ProvideAutoLoadAttributeAnalyzer : DiagnosticAnalyzer
+    {
+        /// <summary>
+        /// The value to use for <see cref="DiagnosticDescriptor.Id"/> in generated diagnostics.
+        /// </summary>
+        public const string Id = "VSSDK004";
+
+        /// <summary>
+        /// A reusable descriptor for diagnostics produced by this analyzer.
+        /// </summary>
+        internal static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
+            id: Id,
+            title: "Auto loads without background load option will be deprecated in future updates.",
```

Comment:
The title should be very short (3-4 words at most) since it shows up in Solution Explorer's analyzer node and otherwise produces very long horizontal scrolling in order to read it.

Reply:
I see you revised the title, but it's still quite long.